shell/platform/tizen/channels/app_control_channel.cc

```diff
@@ -142,41 +143,28 @@ void AppControlChannel::UnregisterReplyHandler() {
   reply_sink_.reset();
 }
 
-template <typename T>
-bool AppControlChannel::GetValueFromArgs(const EncodableValue* args,
-                                         const char* key,
-                                         T& out) {
-  if (std::holds_alternative<EncodableMap>(*args)) {
-    EncodableMap map = std::get<EncodableMap>(*args);
-    if (map.find(EncodableValue(key)) != map.end()) {
-      EncodableValue value = map[EncodableValue(key)];
-      if (std::holds_alternative<T>(value)) {
-        out = std::get<T>(value);
-        return true;
-      }
-    }
-    FT_LOG(Info) << "Key " << key << " not found.";
+std::shared_ptr<AppControl> AppControlChannel::GetAppControl(
+    const EncodableValue* arguments) {
+  auto map_ptr = std::get_if<EncodableMap>(arguments);
+  if (!map_ptr) {
+    FT_LOG(Error) << "Invalid arguments";
+    return nullptr;
   }
-  return false;
-}
 
-std::shared_ptr<AppControl> AppControlChannel::GetAppControl(
-    const EncodableValue* args) {
-  int id;
-  if (!GetValueFromArgs<int>(args, "id", id)) {
+  EncodableMapValueGetter<int> id(map_ptr, "id");
```

Comment:
What's the point? Why do we need an extra holder like this?

Reply:
There's no serious reason.. I've ​bothered that declaring a variable, calling a value extraction function, and checking the return value every time.

Reply:
I'm asking about the difference between using `GetValueFromArgs` and `EncodableMapValueGetter`. As far as I can see using `EncodableMapValueGetter` just makes the code longer, because `EncodableValue` has to be casted to `EncodableMap` and assigned to a variable `map_ptr` every time.

Reply:
No, it doesn't make it longer. rather,  it looks cleaner for me~ 
Even though there are no big differences, EncodableMapValueGetter is checking EncodableMap every time. This is inefficient. It is also efficient that keep the value as the pointer because std::get_if returns value as a pointer.

However, maintaining the template function form and improving only an inefficiency is also a good option.

Reply:
I personally prefer the old method (using a function) in terms of readability for new developers but I don't mind if it looks better for you.